deployments/helm/nvidia-dra-driver-gpu/templates/_helpers.tpl

```diff
@@ -138,3 +138,21 @@ Filter a list by a set of valid values
   {{- end }}
   {{- $result -}}
 {{- end -}}
+
+{{/*
+Get all namespaces (driver namespace + previous driver namespaces from environment variable)
+*/}}
+{{- define "nvidia-dra-driver-gpu.namespaces" -}}
+{{- $namespaces := list (include "nvidia-dra-driver-gpu.namespace" .) }}
+{{- if .Values.controller.containers.computeDomain.env }}
+{{- range .Values.controller.containers.computeDomain.env }}
+{{- if eq .name "PREVIOUS_DRIVER_NAMESPACES" }}
+{{- if .value }}
+{{- $previousNamespaces := splitList "," .value }}
+{{- $namespaces = concat $namespaces $previousNamespaces }}
+{{- end }}
+{{- end }}
+{{- end }}
+{{- end }}
+{{- join "," $namespaces -}}
+{{- end -}}
```

Comment:
This should suffice for the standalone helm chart. We will need to duplicate this logic (creating RBAC objects for each namespace) in the GPU Operator controller. Question -- should we add an explicit field in ClusterPolicy for specifying `PREVIOUS_DRIVER_NAMESPACES` or should our controller just iterate over the list of envvars configured as is done here?

cc @tariq1890

Reply:
I'd be in favour of exposing this knob as a clusterpolicy field/helm value. I understand the view of not wanting to clutter the API; still, I think a zero-downtime migration path is "valid" enough (even if not frequently used) use case to warrant a field where users can specify the additional namespaces. I see this as a relatively low-cost change to the API and it also keeps options open if users ever want to move to a different namespace from that of the operator. It's hard to say right now if a scenario like that would ever come up, but you never know !

Reply:
I am not opposed to adding a field to the ClusterPolicy for this. I am envisioning either `draDriver.computeDomains.additionalNamespaces` or `draDriver.computeDomains.previousDriverNamespaces` , which when set would trigger the GPU Operator controller to create one set of RBAC objects per namespace specified.